homeassistant/components/homekit/accessories.py

```diff
@@ -85,8 +100,32 @@ def update_state_callback(self, entity_id=None, old_state=None,
         _LOGGER.debug('New_state: %s', new_state)
         if new_state is None:
             return
+        self.hass.async_add_job(self.update_battery_level, new_state)
         self.hass.async_add_job(self.update_state, new_state)
 
+    def update_battery_level(self, new_state):
+        """Update battery service if available."""
+        battery_level = convert_to_float(
+            new_state.attributes.get(ATTR_BATTERY_LEVEL))
+        if battery_level is None:
+            return
+        self.char_battery.set_value(battery_level)
+        self.char_low_battery.set_value(battery_level < 20)
+        _LOGGER.debug('%s: Updated battery level to %d', self.entity_id,
+                      battery_level)
+        self.hass.async_add_job(self.update_battery_charging, new_state)
```

Comment:
This won't work!
`update_bettery_level` is a sync function and `async_add_job` must be called from an `async` context. If you still want to do it that way, use `add_job` instead. Even better would be to just combine the to methods.